src/ToDo/Business/ITaskService.cs

```diff
@@ -12,5 +12,7 @@ public interface ITaskService
 
 	ValueTask<IImmutableList<ToDoTask>> GetAsync(TaskList list, CancellationToken ct);
 
-	ValueTask<IImmutableList<ToDoTask>> GetAllAsync(string displayName = "", CancellationToken ct = default);
+	ValueTask<IImmutableList<ToDoTask>> GetAllAsync(CancellationToken ct = default);
+
+	ValueTask<IImmutableList<ToDoTask>> SearchAsync(string displayName = "", CancellationToken ct = default);
```

Comment:
I think the `displayName` should not be optional for a search...
```suggestion
	ValueTask<IImmutableList<ToDoTask>> SearchAsync(string displayName, CancellationToken ct = default);
```

Reply:
See previous comment but if we do keep searchasync, I'd suggest keeping displayName as optional and then adding a "bool? importance = default"  parameter, so we can load the important tasks